mllib/src/main/scala/org/apache/spark/mllib/linalg/Vectors.scala

```diff
@@ -150,6 +150,26 @@ sealed trait Vector extends Serializable {
       toDense
     }
   }
+
+  /**
+   * Find the index of a maximal element.  Returns the first maximal element in case of a tie.
+   * Returns -1 if vector has length 0.
+   */
+  private[spark] def findMax: Int = {
```

Comment:
`findMax` -> `argmax`? I think we can put this in a separate PR or move it to `DenseVector` only. The problem is with sparse vectors, where we need to return an index with value zero if nonzero elements are negative.